### PR TITLE
add envoy package

### PR DESCRIPTION
**Pull request by rawlingsj**
Signed-off-by: James Rawlings <jrawlings@chainguard.dev>

### CODE REVIEW

File: envoy.yaml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+      branch: v${{package.version}}
+      destination: envoy
+
+  - runs: |
+      export JAVA_HOME=/usr/lib/jvm/openjdk
+      mkdir -p /builder/home/.cache/bazel/_bazel_root
+
+      cd envoy
+
+      # these patches are only required for 1.24.0, they are already merged upstream but 1.25.0 is not yet available.
```
Comment (rawlingsj) [inline envoy.yaml:49]: This is a shame but AFAICT necessary until 1.25.0 is available.  FWIW fetching [envoy](https://github.com/envoyproxy/envoy) `main` branch works with the same config but with ALL patches removed.  So this is untidy for now but we can't depend on a main upstream branch and forking for now doesn't seem sensible either.  So using the patch approach.

Comment (kaniini) [reply envoy.yaml:49]: It makes sense, even if I don't like it.

**Review by kaniini — COMMENTED** [review]
I see there are a lot of external deps that it fetches and builds.  Is it possible to use the system-provided deps instead?  This would cut down the package size a lot.

**Comment by rawlingsj** [comment]
> I see there are a lot of external deps that it fetches and builds. Is it possible to use the system-provided deps instead? This would cut down the package size a lot.

I'm not yet sure if / how Bazel handles system-provided deps.  FWIW I did just test the binary size after adding the `-strip:` step and the binary came down from 405M to 58M.

**Comment by rawlingsj** [comment]
I suspect we'd need to start patching the bazel rules and take on the maintenance of them which I guess brings in it's own risks, as we're not running the test suites the envoy project is.

**Review by kaniini — APPROVED** [review]

**Review by kaniini — APPROVED** [review]

**Comment by rawlingsj** [comment]
Maybe there is a way, not too sure TBH.

**Review by puerco — APPROVED** [review]

**Comment by puerco** [comment]
The SBOM on this one is going to be a challenge.

**Comment by dlorenc** [comment]
> I'm not yet sure if / how Bazel handles system-provided deps. FWIW I did just test the binary size after adding the `-strip:` step and the binary came down from 405M to 58M.

This is within range of the official build:
image
https://github.com/envoyproxy/envoy/releases/tag/v1.24.0

**Review by dlorenc — APPROVED** [review]

**Comment by dlorenc** [comment]
> The SBOM on this one is going to be a challenge.

In general for bazel projects this will be challenging. Envoy has done a lot of work to make this easy though. They require manual CPE declaration for all dependencies: https://github.com/envoyproxy/envoy/blob/main/DEPENDENCY_POLICY.md#declaring-external-dependencies

**Pull request merged**